scripts/compare_noupdate_xml_records.py

```diff
@@ -185,20 +196,29 @@ def main(argv=None):
                         del attribs[attr]
                 element.append(
                     etree.Element(record_old_dict[key].tag, attribs))
-            else:
+            elif noupdate:
                 oldrepr = get_node_value(record_old_dict[key])
                 newrepr = get_node_value(record_new_dict[key])
 
                 if oldrepr != newrepr:
                     element.append(deepcopy(record_new_dict[key]))
 
-        for key in record_new_dict.keys():
-            if not record_old.xpath(key):
-                element.append(deepcopy(record_new_dict[key]))
+        if noupdate:
+            for key in record_new_dict.keys():
+                if not record_old.xpath(key):
+                    element.append(deepcopy(record_new_dict[key]))
 
         if len(element):
             data.append(element)
 
+    data.append(etree.Comment("Changes in noupdate data"))
+    for xml_id, record_new in new_noupdate.items():
+        compare_record(xml_id, record_new)
```

Comment:
I'm not assigning old_record, but new_record and I'm passing this to an inline method that adds to the list of elements which is scoped under the main method.